cpp/src/bitmask/bit_mask.h

```diff
@@ -0,0 +1,291 @@
+    return GDF_SUCCESS;
+  }
+
+  /**
+   *  @brief Deallocate device space for the valid bit mask
+   *
+   *  @param[in]  valid   The pointer to device space that we wish to deallocate
+   *
+   *  @return GDF_SUCCESS on success, the CUDA error on failure
+   */
+  inline gdf_error destroy_bit_mask(bit_mask_t *valid) {
+    RMM_TRY(RMM_FREE(valid, 0));
+    return GDF_SUCCESS;
+  }
+
+  /**
+   *  @brief Get a single element of bits from the device
+   *
+   *  @param[out]  element - address on host where the bits will be stored
+   *  @param[out]  device_element - address on the device containing the bits to fetch
+   *
+   *  @return GDF_SUCCESS on success, the CUDA error on failure
+   */
+  inline gdf_error get_element(bit_mask_t *element, const bit_mask_t *device_element) {
+    CUDA_TRY(cudaMemcpy(element, device_element, sizeof(bit_mask_t), cudaMemcpyDeviceToHost));
+    return GDF_SUCCESS;
+  }
+	  
+  /**
+   *  @brief Put a single element of bits to the device
+   *
+   *  @param[out]  element - address on host containing bits to store
+   *  @param[out]  device_element - address on the device where the bits will be stored
+   *
+   *  @return GDF_SUCCESS on success, the CUDA error on failure
+   */
+  inline gdf_error put_element(bit_mask_t element, bit_mask_t *device_element) {
+    CUDA_TRY(cudaMemcpy(device_element, &element, sizeof(bit_mask_t), cudaMemcpyHostToDevice));
+    return GDF_SUCCESS;
+  }
+
+  /**
+   *  @brief Allocate device space for the valid bit mask.
+   *
+   *  @param[out] mask                  address of the bit mask pointer
+   *  @param[in]  number_of_records     number of records
+   *  @param[in]  fill_value            optional, should the memory be initialized to all 0 or 1s. All other
+   *                                    values indicate un-initialized.  Default is uninitialized
+   *  @param[in]  padding_bytes         optional, specifies byte boundary the data should be padded to.
+   *                                    Defaults to 64 bytes, meaning the space allocated will be rounded
+   *                                    up to the next multiple of 64 bytes.
+   *
+   *  @return GDF_SUCCESS on success, the RMM or CUDA error on error
+   */
+  gdf_error create_bit_mask(bit_mask_t **mask, gdf_size_type number_of_records, int fill_value = -1, gdf_size_type padding_bytes = 64) {
+    //
+    //  To handle padding, we will round the number_of_records up to the next padding boundary, then identify how many element
+    //  that equates to.  Then we can allocate the appropriate amount of storage.
+    //
+    gdf_size_type num_bytes = (number_of_records + 7) / 8;
+    gdf_size_type num_padding_blocks = (num_bytes + padding_bytes - 1) / padding_bytes;
+    gdf_size_type num_elements = bit_mask::num_elements(num_padding_blocks * 8 * padding_bytes);
+
+    RMM_TRY(RMM_ALLOC(mask, sizeof(bit_mask_t) * num_elements, 0));
+
+    if (fill_value == 0) {
+      //
+      //  Padding with zero is an easy case
```

Comment:
I think you mean Filling with zero.
```suggestion
      //  Filling with zero is an easy case
```

Reply:
Actually the comment is completely unnecessary here (waste of 3 LOCs), could delete it without loss of clarity.